### PR TITLE
Fix kube-inject with no webhook having errors

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
When there is no webhook exists, kube-inject will fall back to use injector config maps. However the values parameter is not set ahead, which leads to the error `Error: failed to run injection template: failed parsing generated injected YAML (check Istio sidecar injector configuration): unmarshal patched pod: json: cannot unmarshal string into Go struct field Probe.spec.containers.readinessProbe.failureThreshold of type int32`.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/cmd/kubeinject_test.go [hunk 1]
```diff
@@ -87,3 +95,59 @@ func TestKubeInject(t *testing.T) {
 		})
 	}
 }
+
+func TestKubeInjectWithoutWebhook(t *testing.T) {
```
Comment (ramaraochavali) [inline istioctl/cmd/kubeinject_test.go:99]: Is it possible to fold this test as an additional test case in the above test?

Comment (hanxiaop) [reply istioctl/cmd/kubeinject_test.go]: Revised.

File: releasenotes/notes/41326.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** an issue that makes `kube-inject` not working as expected when there is no webhook exists.
```
Comment (ericvn) [inline releasenotes/notes/41326.yaml:6]:
```suggestion
    **Fixed** an issue where `kube-inject` does not work as expected when no webhook exists.
```

**Comment by hanxiaop** [comment]
not stale

**Comment by therealmitchconnors** [comment]
Can you describe the root cause and solution?  I am having trouble seeing how this code relates to the linked issue...

**Comment by hanxiaop** [comment]
> Can you describe the root cause and solution? I am having trouble seeing how this code relates to the linked issue...

@therealmitchconnors The root cause is: when we have the webhook injector, `kube-inject` will use the injector to inject the file. However if there is no webhook, `kube-inject` falls back to use configs from the injection configmap. In this situation, the old logic is missing the part to obtain the `values` config, which leads to unmarshalling errors. This PR is to get and set `values` config in this situation, and make `kube-inject` work properly.

**Comment by hanxiaop** [comment]
not stale

**Pull request closed without merging**